File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    # key : 선수이름, value : 등수로 딕셔너리 초기화
    player_dict = {player:idx for idx, player in enumerate(players)}
    
    for c in callings:
        idx = player_dict[c]
        players[idx], players[idx- 1] = players[idx-1], players[idx]
        
        player_dict[players[idx]] = idx
        player_dict[players[idx-1]] = idx -1
                
    
    return players

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.java
import java.util.*;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        Map<String, Integer> map = new HashMap();
        
        for (int i = 0; i< players.length; i++){
            map.put(players[i], i);
        }
        
        for (int i = 0; i< callings.length; i++){
            String name = callings[i];
            int index = map.get(name);
            
            String temp = players[index - 1];
            players[index - 1] = players[index];
            players[index] = temp;
            
            map.put(name, index-1);
            map.put(temp, index);
        }
        return players;
    }   
}